src/Particle/createDistanceTableAB.cpp

```diff
@@ -25,14 +25,14 @@ namespace qmcplusplus
  *\param s source/target particle set
  *\return index of the distance table with the name
  */
-DistanceTableData* createDistanceTableAB(const ParticleSet& s, ParticleSet& t, std::ostream& description)
+DistanceTable* createDistanceTableAB(const ParticleSet& s, ParticleSet& t, std::ostream& description)
```

Comment:
why not a smart pointer?

These need to throw an exception if they fail, when it happens it will frequently be in another classes a constructor.

Reply:
I also caught this but consider this a follow-up topic as more places needs to be touched.

Reply:
Added #3616 to remember it.